test_rclcpp/test/node_check_names.cpp

```diff
@@ -42,11 +42,21 @@ int main(int argc, char ** argv)
   const std::chrono::steady_clock::time_point max_runtime =
     std::chrono::steady_clock::now() + std::chrono::seconds(10);
 
+  std::vector<bool> found(num_nodes);
+
   while (rclcpp::ok()) {
     auto names = node->get_node_graph_interface()->get_node_names();
     for (auto it : names) {
       if (it.compare(0, node_to_look_for.length(), node_to_look_for) == 0) {
-        counter++;
+        try {
+          int idx = std::stoi(it.substr(node_to_look_for.length()));
+          if (!found[idx]) {
+            found[idx] = true;
+            counter++;
```

Comment:
you are correct, this check was broken before.

i would suggest to remove `counter` variable here, instead we can check if found is all true.

Reply:
I might be mistaken, but the logic seems complete to me with better efficiency. Since `found` array is properly initialized, and the counter is incremented only when that idx is not found.

Reply:
i did not mean the PR is broken, but can remove `counter` with checking the `found` array to make sure all elements are true?

Reply:
Okay. Here's a version without `counter` variable. 

```cpp
int idx = std::stoi(it.substr(node_to_look_for.length()));
if (!found[idx]) {
  found[idx] = true;
  bool all_found = true;
  for (auto v: found) {
    if (!v) {
      all_found = false;
      break;
    }
  }
  if (all_found) {
    break;
  }
}
```

But I think this is lengthy and not efficient. What do you think? 

Moreover, we have error information at the end of the test.
```cpp
exec.remove_node(node);
rclcpp::shutdown();
if (counter < num_nodes) {
fprintf(stderr, "Did not find all %d nodes\n", num_nodes);
fprintf(stderr, "Found %d nodes\n", counter);
return 1;
}
return 0;
```
Would you like to remove this, too? Or we iterate the `found` array again.

Reply:
either looks good to me, i was just thinking that we could remove redundant variables.